src/handlers/LedgerSigner.ts

```diff
@@ -0,0 +1,188 @@
+'use strict';
+
+import AppEth, { ledgerService } from '@ledgerhq/hw-app-eth';
+import TransportBLE from '@ledgerhq/react-native-hw-transport-ble';
+import { SignTypedDataVersion, TypedDataUtils } from '@metamask/eth-sig-util';
+import { ethers } from 'ethers';
+import { logger, RainbowError } from '@/logger';
+
+function waiter(duration: number): Promise<void> {
+  return new Promise(resolve => {
+    setTimeout(resolve, duration);
+  });
+}
+
+export class LedgerSigner extends ethers.Signer {
+  readonly path: string | undefined;
+  readonly privateKey: null | undefined;
+  readonly deviceId: string | undefined;
+  readonly isLedger: boolean | undefined;
+
+  readonly _eth: Promise<AppEth> | undefined;
+
+  constructor(
+    provider: ethers.providers.Provider,
+    path: string,
+    deviceId: string
+  ) {
+    super();
+
+    ethers.utils.defineReadOnly(this, 'isLedger', true);
+    ethers.utils.defineReadOnly(this, 'privateKey', null);
+    ethers.utils.defineReadOnly(this, 'path', path);
+    ethers.utils.defineReadOnly(this, 'deviceId', deviceId);
+    ethers.utils.defineReadOnly(this, 'provider', provider || null);
+    ethers.utils.defineReadOnly(
+      this,
+      '_eth',
+      TransportBLE.open(deviceId).then(
+        transport => {
+          const eth = new AppEth(transport);
+          return eth;
+        },
+        error => {
+          return Promise.reject(error);
+        }
+      )
+    );
+  }
+
+  // @skylarbarrera - may end up removing/tweaking retry logic but for now it works and lets us move forward
+  _retry<T = any>(
+    callback: (eth: AppEth) => Promise<T>,
+    timeout?: number
+  ): Promise<T> {
+    return new Promise(async (resolve, reject) => {
+      if (timeout && timeout > 0) {
+        setTimeout(() => {
+          logger.debug(
+            'Ledger: Signer timeout',
+            {},
+            logger.DebugContext.ledger
+          );
+          return reject(new RainbowError('Ledger: Signer timeout'));
+        }, timeout);
+      }
+
+      const eth = await this._eth;
+      if (!eth) {
+        logger.debug(
+          'Ledger: Eth app not open',
+          {},
+          logger.DebugContext.ledger
+        );
+        return reject(new Error('Ledger: Eth app not open'));
```

Comment:
I see your comment, but let's make sure to come back and think about how we want to handle errors in here. Like, atm none of these will be sent to Sentry unless we wrap the method calls in a try/catch, and we might retry up to 50 times, which sounds like overkill to me.

We def want to send errors to Sentry from here, and then where we implement this class in our code we should probably look for specific error instances or codes e.g. `LedgerTrasportLockedError` in order to better handle the failures of this system.

Reply:
We might think about `_retry` here as more of a generic "do X safely" with optional params of `retryCount` and `timeout`.